plugins/lang-plan-a/src/main/java/org/elasticsearch/plan/a/Analyzer.java

```diff
@@ -254,29 +247,36 @@ public Void visitIf(final IfContext ctx) {
 
         final BlockContext blockctx0 = ctx.block(0);
         final StatementMetadata blocksmd0 = adapter.createStatementMetadata(blockctx0);
-        blocksmd0.last = ifsmd.last;
+        blocksmd0.lastSource = ifsmd.lastSource;
```

Comment:
Maybe a method on `StatementMetdata` to copy the appropriate metadata. It saves a few lines of code and forces you to name the operation.